src/schemas/json/rust-project.json

```
@@ -0,0 +1,126 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "definitions": {
+    "crate": {
+      "type": "object",
+      "properties": {
+        "display_name": {
+          "type": "string",
+          "description": "Crate name used for display purposes, without affecting semantics."
+        },
+        "root_module": {
+          "type": "string",
+          "description": "Path to the root module of the crate."
+        },
+        "edition": {
+          "type": "string",
+          "description": "Edition of the crate.",
+          "enum": ["2015", "2018", "2021"]
+        },
+        "deps": {
+          "type": "array",
```

```suggestion
          "type": "array",
          "uniqueItems": true,
```

Optional fix (available in multiple places): deny item repetition when it's possible (when duplicates are not allowed).